driver-core/src/main/com/mongodb/internal/connection/DefaultConnectionPool.java

```diff
@@ -1416,18 +1417,21 @@ private BackgroundMaintenanceManager() {
         void start() {
             if (maintainer != null) {
                 assertTrue(cancellationHandle == null);
-                cancellationHandle = maintainer.scheduleAtFixedRate(DefaultConnectionPool.this::doMaintenance,
+                cancellationHandle = doIgnoringRejectedExecution(() -> maintainer.scheduleAtFixedRate(
+                        DefaultConnectionPool.this::doMaintenance,
                         initialStart ? settings.getMaintenanceInitialDelay(MILLISECONDS) : 0,
-                        settings.getMaintenanceFrequency(MILLISECONDS), MILLISECONDS);
+                        settings.getMaintenanceFrequency(MILLISECONDS), MILLISECONDS));
                 initialStart = false;
             }
         }
 
         void runOnceAndStop() {
             if (maintainer != null) {
-                assertNotNull(cancellationHandle).cancel(false);
-                cancellationHandle = null;
-                maintainer.execute(DefaultConnectionPool.this::doMaintenance);
+                if (cancellationHandle != null) {
```

Comment:
What's the execution path where cancellationHandle can be null?  I'm not sure why it's no longer an assertion.

Reply:
When `BackgroundMaintenanceManager.start` is called concurrently with `close`, it may fail to schedule `doMaintenance`, writing `null` to `cancellationHandle` as a consequence. `BackgroundMaintenanceManager.runOnceAndStop` called after `start` (these methods must be called sequentially as per the documentation of `BackgroundMaintenanceManager`) observes `null` in `cancellationHandle`.